### PR TITLE
Swagger for Machine API

### PR DESCRIPTION
**Pull request by sushmak295**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [public repo review queue], [private repo review queue]  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue]  

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [x] Other, please clarify:
    - _new machine list, get api proposal: https://msazure.visualstudio.com/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/430142/Exposing-Machine-Level-Information(Ip-Host-Name)-in-an-Agentpool_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[public repo review queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo review queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Review by FumingZhang — COMMENTED** [review]
Please fix sematic errors

> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/parameters/AgentPoolNameParameter' is not referencing a valid location. paths,/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/machines,get,parameters,4
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1385:11
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/definitions/MachineListResult' is not referencing a valid location. paths,/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/machines,get,responses,200,schema
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1392:13
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/parameters/AgentPoolNameParameter' is not referencing a valid location. paths,/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/machines/{machineName},get,parameters,4
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1433:11
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/parameters/MachineNameParameter' is not referencing a valid location. paths,/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/machines/{machineName},get,parameters,5
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1436:11
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/definitions/Machine' is not referencing a valid location. paths,/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/machines/{machineName},get,responses,200,schema
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1443:13
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/definitions/Machine' is not referencing a valid location. parameters,MachineListResult,properties,value,items
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:8596:11
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/definitions/MachineProperties' is not referencing a valid location. parameters,Machine,properties,properties
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:8617:9
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/definitions/MachineNetworkProperties' is not referencing a valid location. parameters,MachineProperties,properties,network
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:8629:9
> cmderr	[generator automation-v2] [ERROR] error   | InvalidRef | Ref '#/definitions/MachineIpAddress' is not referencing a valid location. parameters,MachineNetworkProperties,properties,ipAddresses,items
> cmderr	[generator automation-v2] [ERROR] - file:///mnt/vss/_work/1/s/azure-rest-api-specs/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:8648:11

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineGet.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                                "family": "IPv4"
+                            }
+                        ]
+                    },
+                    "resourceId": "/subscriptions/26ad903f-2330-429d-8389-864ac35c4350/resourceGroups/MC_skuchipuebld74360722_skuchipudiVMSSPrivate_eastus/providers/Microsoft.Compute/virtualMachineScaleSets/aks-nodes-41768208-vmss/virtualMachines/1"
+                }
+            }
+        }
+    }
+}
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineGet.json:34]: minor: missing newline at EOF

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineList.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                },
+                "resourceId": "/subscriptions/26ad903f-2330-429d-8389-864ac35c4350/resourceGroups/MC_skuchipuebld74360722_skuchipudiVMSSPrivate_eastus/providers/Microsoft.Compute/virtualMachineScaleSets/aks-nodes-41768208-vmss/virtualMachines/1"
+              }
+            }
+          ],
+          "nextLink": "http://xxxx.azure.com?encodedToken=c2tpcFRva2VuPTE"
+        }
+      }
+    }
+  }
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineList.json:38]: minor: missing newline at EOF

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json [hunk 3]
```diff
@@ -3869,6 +3929,104 @@
       },
       "description": "Agent Pool."
     },
+    "MachineListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:3936]: Please add the identifier field

```suggestion
          "type": "array",
          "x-ms-identifiers": ["id"],
```

**Comment by FumingZhang** [comment]
Hey @sushmak295, could you please fix the failed CI [Swagger LintDiff](https://github.com/Azure/azure-rest-api-specs/pull/25180/checks?check_run_id=15697232713)?

image

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json [hunk 4]
```diff
@@ -3869,6 +3929,104 @@
...
+          "items": {
+            "$ref": "#/definitions/Machine"
+          },
+          "description": "The list of Machines in cluster."
+        },
+        "nextLink": {
+          "type": "string",
```
Comment (serbrech) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:3943]: add `"format": "uri",`

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineGet.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "subscriptionId": "subid1",
+    "resourceGroupName": "rg1",
+    "resourceName": "clustername1",
+    "agentPoolName": "agentpool1",
+    "machineName": "aks-nodepool1-42263519-vmss00000t"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/dummyHostName",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineGet.json:13]: ID should include a fully specified resourceID, similar to the one below in `resourceId`

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineList.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    "resourceGroupName": "rg1",
+    "resourceName": "clustername1",
+    "agentPoolName": "agentpool1"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/dummyHostName",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineList.json:14]: Same comment

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineGet.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        "name": "dummyHostName",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineGet.json:14]: Can you get an actual example hostname? So users understand roughly what to expect?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineList.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            "name": "dummyHostName",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/examples/MachineList.json:15]: Same, actual hostname?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json [hunk 9]
```diff
@@ -1362,6 +1333,102 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/machines/{machineName}": {
+      "get": {
+        "tags": [
+          "Machine"
+        ],
+        "operationId": "Machine_Get",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1389]: Should be `Machines_Get`

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json [hunk 10]
```diff
@@ -1362,6 +1333,102 @@
...
+          "List Machines in an Agentpool by Managed Cluster": {
+            "$ref": "./examples/MachineList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/machines/{machineName}": {
+      "get": {
+        "tags": [
+          "Machine"
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1387]: Machines

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json [hunk 11]
```diff
@@ -1362,6 +1333,102 @@
...
+        ],
+        "operationId": "Machine_Get",
+        "summary": "Get info of spec machine in the specified managed cluster adn specified agentpool.",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1390]:
```suggestion
        "summary": "Get a specific machine in the specified agent pool.",
```

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json [hunk 12]
```diff
@@ -1553,9 +1620,6 @@
           }
         ],
         "responses": {
-          "204": {
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:1556]: Why reorder this?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json [hunk 13]
```diff
@@ -3869,6 +3929,109 @@
...
+          "$ref": "#/definitions/MachineProperties",
+          "description": "The properties of the machine"
+        }
+      },
+      "description": "A machine. Contains details about the underlying virtual machine. A machine may be visible here but not in kubectl get nodes; if so it may be because the machine has not been registered with the Kubernetes API Server yet."
+    },
+    "MachineProperties": {
+      "type": "object",
+      "readOnly": true,
+      "description": "properties of the machine",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json:3972]:
```suggestion
      "description": "The properties of the machine",
```

**Review by FumingZhang — APPROVED** [review]

**Pull request merged**